File: 프로그래머스/lv1/87389. 나머지가 1이 되는 수 찾기/나머지가 1이 되는 수 찾기.js
function solution(n) {
    // n%x === 1
    // 가장 작은 x를 구해야한다.
    for(let i=n; i>=0; i--){
        if(n%(i-n) === 1) {
            return (n-i);
        }
    }
}
/*
    접근법 :
    n%0, 1, 2, 3, ... 나누는 값을 작은 수 부터 올라가며 대입해주면서
    === 1일 때 바로 리턴이 되게끔 만들어 주었다.

    
    다른 접근법 :
    functrion solution(n) {

        for(int i=1; i<n; i++) {
            if(n%i === 1) { return i; }
        }
        
    }
*/                

File: 프로그래머스/lv1/87389. 나머지가 1이 되는 수 찾기/나머지가 1이 되는 수 찾기.java
class Solution {
    public int solution(int n) {
        
        for(int i=n-1; i>0; i--) {
            
            if(n%(n-i) == 1) {
                return (n-i);
            }
            
        }
        return n;
    }
}
/*
    접근법 :
    내가 접근한 방법은 너무 돌아간? 접근인것 같아
    다른 접근법을 중심으로 기록하려한다. ㅠㅠ


    다른 접근법 :
    class Solution {
        public int solution(int n) {

            // 2보다 작은수로 나누면 오류가 난다 ㅠㅠ
            for(int i=2; i<n; i++) {
                if(n%i == 1) { return i; }
            }

        }
        // 모든 상황을 커버해야함, 그냥 들어온 수 나가게 만들어줌
        return n;

    } // end class
*/
